lib/puppet/provider/cisco_interface/cisco.rb

```diff
@@ -185,19 +186,17 @@ def self.properties_get(interface_name, nu_obj, all_intf: nil)
     new(current_state)
   end # self.properties_get
 
-  def self.instances(single_intf=nil)
+  def self.instances(single_intf=nil, interface_threshold=0)
     # 'puppet resource' calls here directly; will always get all interfaces.
     # 'puppet agent' callpath is initialize->prefetch; may pass a single intf.
-    all_intf = single_intf ? false : true
-    interfaces = []
-    if Gem::Version.new(CiscoNodeUtils::VERSION) > Gem::Version.new('2.0.2')
+    if single_intf && interface_threshold > 0
+      all_intf = false
       nu_interfaces = Cisco::Interface.interfaces(nil, single_intf)
     else
-      info '## Notice: Unable to prefetch independently, using legacy lookup instead.'
```

Comment:
Why did these first two notices not make it into the method above?  I feel like these add good information and would prefer we keep them.

Reply:
They live in `cmnutils` now. I moved them there so that I wouldn't need to specify them in each provider.